src/matchers/root.ts

```diff
@@ -0,0 +1,6 @@
+import { Node } from "typescript";
+import { TSQuerySelectorNode } from "../tsquery-types";
+
+export function  root (node: Node, selector: TSQuerySelectorNode, ancestry: Array<Node>): boolean {
+    return ancestry.length === 0;
```

Comment:
This work only if we are in the root of the tree. I think when the `:root` selector is matched the node selection should be rebased on the `sourceFileObject`.